financial-services-accelerator/components/org.wso2.financial.services.accelerator.event.notifications.service/src/main/java/org/wso2/financial/services/accelerator/event/notifications/service/RealtimeNotificationService.java

```diff
@@ -0,0 +1,142 @@
+            throws FSEventNotificationException {
+
+        EventNotificationDAO eventNotificationDAO = EventNotificationStoreInitializer.getEventNotificationDAO();
+
+        Connection connection = DatabaseUtils.getDBConnection();
+
+        try {
+            List<Notification> events = eventNotificationDAO.
+                    getNotificationsByStatus(connection, status);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Event Notifications with %s status retrieved successfully.",
+                        status.replaceAll("[\r\n]", "")));
+            }
+            DatabaseUtils.commitTransaction(connection);
+            return events;
+        } catch (FSEventNotificationException e) {
+            log.error("Error while retrieving event notification.", e);
+            DatabaseUtils.rollbackTransaction(connection);
+            throw new FSEventNotificationException(EventNotificationConstants.ERROR_STORING_EVENT_SUBSCRIPTION, e);
+        } finally {
+            log.debug(EventNotificationConstants.DATABASE_CONNECTION_CLOSE_LOG_MSG);
+            DatabaseUtils.closeConnection(connection);
+        }
+    }
+
+    /**
+     * Method to retrieve notifications by NotificationID.
+     *
+     * @param notificationId  Notification ID to retrieve
+     * @return List of notifications by notification ID
+     * @throws FSEventNotificationException  Exception when retrieving notifications by notification ID
+     */
+    public List<NotificationEvent> getEventsByNotificationID(String notificationId)
+            throws FSEventNotificationException {
+
+        EventNotificationDAO eventNotificationDAO = EventNotificationStoreInitializer.getEventNotificationDAO();
+
+        Connection connection = DatabaseUtils.getDBConnection();
+
+        try {
+            //store event subscription data in the database
+            List<NotificationEvent> notificationEvents = eventNotificationDAO.
+                    getEventsByNotificationID(connection, notificationId);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Event Notifications with notification id %s retrieved successfully.",
+                        notificationId.replaceAll("[\r\n]", "")));
+            }
+            DatabaseUtils.commitTransaction(connection);
+            return notificationEvents;
+        } catch (FSEventNotificationException e) {
+            log.error("Error while retrieving event notification.", e);
+            DatabaseUtils.rollbackTransaction(connection);
+            throw new FSEventNotificationException(EventNotificationConstants.ERROR_STORING_EVENT_SUBSCRIPTION, e);
+        } finally {
+            log.debug(EventNotificationConstants.DATABASE_CONNECTION_CLOSE_LOG_MSG);
+            DatabaseUtils.closeConnection(connection);
+        }
+    }
+
+    /**
+     * Method to update the notification status by ID, allowed values are.
+     * OPEN,ACK and ERR
```

Comment:
if only these values are allowed shouldn't we accept an enum?

Reply:
FIxed in https://github.com/wso2/financial-services-accelerator/pull/193/commits/00babc11f6c634ec7023e46cba1a28b73d4377e7